### PR TITLE
fix: add v2 skus to api-management/service

### LINKED ISSUE #1615: [AVM Module Issue]: API Management Service - Please support API Management V2 SKUs
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/api-management/service

### (Optional) Module Version

_No response_

### Description

Can you add support for API Management V2 SKUs? We really want to use the new V2 SKU options, but are currently forced to write our own bicep instead of using the awesome AVM module.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by tony-box**
## Description

Adding capability for v2 APIM skus

Fixes #1615 

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.api-management.service](https://github.com/tony-box/bicep-registry-modules/actions/workflows/avm.res.api-management.service.yml)  |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/avm_module_issue.yml
Comment (eriqua) [inline .github/ISSUE_TEMPLATE/avm_module_issue.yml:1]: Hey @tony-box security center changes sneaked in. Let's have a clean diff including only apim changes.

**Comment by AlexanderSehr** [comment]
oh my @tony-box, what happened? 😄

**Comment by tony-box** [comment]
@AlexanderSehr oh man, I think the stupid auto-merge thing in github did this. I was attempting to update my fork's v2sku branch to the latest commit from upstream so that it would resolve the accidental commits that snuck in related to security-center.  I should have just used my tried and true CLI. Let me sort this out and reopen the PR. Sorry about that!

**Comment by AlexanderSehr** [comment]
Github does force pushes? 😅  Let me know if I can help - especially if you lost progress. 👀

**Comment by tony-box** [comment]
When your fork or branch is out of date from upstream, there's an option in the UI to sync your fork with the upstream. If there are merge conflicts, it'll let you to do something like "revert changes" or "discard commits"--cant remember exactly. I think when I clicked one of those it force pushed to my fork to get rid of all my changes.

I've got it under control though. Updating my branch now :)

**Comment by tony-box** [comment]
I recreated this PR as #1777 just to keep everything clean

**Pull request closed without merging**